test/lib/data/addBudget.firebase.test.ts

```diff
@@ -0,0 +1,120 @@
+import {assert, beforeAll, it, describe, expect} from 'vitest';
+import {getFirestore} from 'firebase/firestore';
+import {Budget, DataModel, User} from 'lib/data';
+import {Collections} from 'lib/firebase';
+import {getLocalFirebase} from '../../utils/database.util';
+
+const db = getFirestore();
+const database = getLocalFirebase(db);
+
+beforeAll(async () => {
+  const testUser: User[] = [1, 2, 3].map(number => {
+    return {
+      id: `test_user${number}`,
+      total_budget: 2000,
+      remaining_budget: 1000,
+      pending_event: 10,
+      completed_event: 10,
+      planned_event: 10,
+      user_name: 'test_user1',
+      user_type: 'RSO',
+    };
+  });
+
+  testUser.push({
```

Comment:
There's already a test user with the `id` of `'test_user1`.  Maybe change the id of this test user to `test_user4`?